include/deal.II/base/patterns.h

```diff
@@ -661,6 +662,139 @@ namespace Patterns
   };
 
 
+
+  /**
+   * This pattern matches comma-separated values of arbitrary types. Each type
+   * has to match a pattern given to the constructor.
+   *
+   * An example usage is the following:
+   *
+   * @code
+   * std::vector< std::unique_ptr<Patterns::PatternBase> > ps;
+   *
+   * ps.push_back(std::unique_ptr<Patterns::Integer>());
+   * ps.push_back(std::unique_ptr<Patterns::Double>());
+   * ps.push_back(std::unique_ptr<Patterns::Anything>());
+   *
+   * Patterns::Tuple pattern(ps, ";");
+   *
+   * bool check = ps.match("5; 3.14; Ciao"); // check = true
+   * @endcode
+   *
+   * The constructor expects a vector of Patterns, and optionally a string
+   * specifying the separator to use when parsing the Tuple from a string.
+   *
+   * @author Luca Heltai, 2017.
+   */
+  class Tuple : public PatternBase
+  {
+  public:
+    /**
+     * Constructor. Use a vector of unique pointers to Patterns to construct
+     * the tuple.
+     *
+     * @param patterns The pattern each object of the Tuple should match
+     * @param separator An optional string used to delimit each element
+     * Constructor.
+     */
+    Tuple (const std::vector<std::unique_ptr<PatternBase> > &patterns,
+           const std::string  &separator = ",");
+
+    /**
+     * Constructor. Creates a Tuple from more than one class derived from
+     * PatternBase.
+     *
+     * @param separator What separator to use.
+     * @param patterns The list of patterns to use
+     */
+    template<class... PatternTypes>
+    Tuple (const std::string  &separator,
+           const PatternTypes &... patterns);
+
+    /**
+     * Constructor. This is needed to allow users to specify
+     * directly the separator without using std::string(";").
+     *
+     * Since we support a pure variadic templates version, without this
+     * specialization, the compiler will fail with criptyc errors.
```

Comment:
Hmm... I am pretty sure this should be `cryptic`

Reply:
ahahahah.

Yes, of course. :D

Reply:
I'll fix this with the next commit. Give me 5 minutes.